integration-tests/tests/src/tests/sync/open.ts

```diff
@@ -0,0 +1,38 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import { expect } from "chai";
+
+import { authenticateUserBefore, importAppBefore } from "../../hooks";
+
+describe("Realm.open on a sync Realm", () => {
+  importAppBefore("with-db-flx");
+  authenticateUserBefore();
+
+  it("fails gracefully when canceled while opening", async function (this: UserContext) {
```

Comment:
These tests seem to fail consistently when I run electron tests locally. I'm not sure but could it be a race condition between the realm opening and the `cancel()` method being invoked? :thinking: 

<img width="1578" alt="image" src="https://user-images.githubusercontent.com/44238293/220147498-e5664b22-1ff9-4bf4-be34-685123691da1.png">